src/Cmd.ml

```diff
@@ -50,18 +50,29 @@ let output ~args ?cwd ?stdin { cmd; env } =
   let shellString =
     Js.Array.joinWith " " (Js.Array.concat args [| Path.toString cmd |])
   in
-  Js.Console.info shellString;
   let cwd =
     match cwd with
     | None -> None
     | Some cwd -> Some (Path.toString cwd)
   in
+  let message =
+    let message =
+      [ ("cmd", Log.field (Path.toString cmd))
+      ; ("args", Log.field (Js.Array.joinWith " " args))
+      ]
+    in
+    match cwd with
+    | None -> message
+    | Some cwd -> ("cwd", Log.field cwd) :: message
+  in
   ChildProcess.exec shellString ?stdin (ChildProcess.Options.make ?cwd ~env ())
-  |> Promise.map (function
-       | ChildProcess.{ exitCode = 0; stdout; stderr = _ } -> Ok stdout
-       | ChildProcess.{ exitCode; stdout = _; stderr } ->
-         Error
-           {j| Command $shellString failed:
-exitCode: $exitCode
-stderr: $stderr
-|j})
+  |> Promise.map (fun (res : ChildProcess.return) ->
+         let () =
+           let message = ("result", Log.field res) :: message in
+           logJson "external command" message
+         in
```

Comment:
What's the purpose of making this a let expression? Wouldn't a semicolon suffice?

Reply:
See the new code. The purpose of the `let ()` is to avoid leaking the `message` binding.